### PR TITLE
Fixes integration tests affected by recent change

### PR DESCRIPTION
**Pull request by dnahodil**

### CODE REVIEW

File: test/integration/au/org/emii/aatams/detection/DetectionFactoryServiceIntegrationTests.groovy [hunk 1]
```diff
@@ -12,15 +12,15 @@ class DetectionFactoryServiceIntegrationTests extends GroovyTestCase {
...
-
         def sql = new Sql(dataSource)
         ValidDetection.metaClass.static.count = {
 
             return sql.firstRow('select count(*) from valid_detection').count
         }
 
+        def initValidCount = ValidDetection.count()
+        def initInvalidCount = InvalidDetection.count()
+
```
Comment (dnahodil) [inline test/integration/au/org/emii/aatams/detection/DetectionFactoryServiceIntegrationTests.groovy]: I had to move these calls to count() to after we add our own version of count() to the ValidDetection metaclass.

**Comment by jkburges** [comment]
Tests all work now?

**Comment by dnahodil** [comment]
Yep

**Comment by dnahodil** [comment]
Well.

Apart from QueryServiceTests testBetweenRestriction but that one is a bit special.

**Pull request merged**